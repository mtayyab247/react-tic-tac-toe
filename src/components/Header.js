import React from 'react';
import '../css/header.css';

const Header = ({playerTurn}) => {

    return(
        <div className="header">
            <div className="title-wrapper">
                <h3 className="title">Enjoy Tic Tac Toe</h3>
            </div>

            <div className="turn-info">
                <div className="turn">{playerTurn}</div>
            </div>
        </div>
    );
};

export default Header;