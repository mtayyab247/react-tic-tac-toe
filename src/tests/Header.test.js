import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Header from '../components/Header';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({
    adapter: new Adapter()
});

describe("<Header /> test", () => {
    it("should render title and player name whose turn it is", () => {
        const wrapper = shallow(<Header playerTurn={"Player 1"} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should not render player turn if playturn prop not passed", () => {
        const wrapper = shallow(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});


