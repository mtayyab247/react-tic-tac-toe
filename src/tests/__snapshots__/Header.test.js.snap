// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> test should not render player turn if playturn prop not passed 1`] = `
<div
  className="header"
>
  <div
    className="title-wrapper"
  >
    <h3
      className="title"
    >
      Enjoy Tic Tac Toe
    </h3>
  </div>
  <div
    className="turn-info"
  >
    <div
      className="turn"
    />
  </div>
</div>
`;

exports[`<Header /> test should render title and player name whose turn it is 1`] = `
<div
  className="header"
>
  <div
    className="title-wrapper"
  >
    <h3
      className="title"
    >
      Enjoy Tic Tac Toe
    </h3>
  </div>
  <div
    className="turn-info"
  >
    <div
      className="turn"
    >
      Player 1
    </div>
  </div>
</div>
`;
